core/src/main/scala/org/apache/spark/sql/internal/JarUtils.scala

```diff
@@ -98,6 +116,35 @@ object ContextJarUtils extends Logging {
     if (!jarDirectory.exists()) jarDirectory.mkdir()
     jarDirectory
   }
+
+  def addToTheListInCmdRegion(k: String, item: String, head: String): Unit = {
```

Comment:
I won't suggest to take care of making it transactional but at least put in the docs that parallel 'Create/Drop Function' from different threads should not be done.